File: solved.ac/python/2312.js
const fs = require('fs')


let input = fs.readFileSync('./dev/stdin', 'utf8')

input = input.split('\r\n')

let index = 1

for (let i = 0 ; i < parseInt(input[0]); i++) {
  let ans = []
  let num = parseInt(input[index])
  let temp = num
  if (!num) break


  for (let j = 2 ; j <= num; j++) {
    while ( num % j == 0 ) {
      ans.push(j)
      num = num / j
    }
  }
  for (let k = 2 ; k <= temp; k++) {
    if (ans.indexOf(k) != -1) {
      let l = ans.filter(v => v == k).length
      console.log(`${k} ${l}`)
    }
  }


  index++
}

File: solved.ac/python/2312.py
t = int(input())

# 소인수 분해한 결과를 출력하는 프로그램을 작성하시오
def solution(n):
  temp = n
  ans = []
  for i in range(2, n+ 1):
    while n % i == 0:
      ans.append(i)
      n = n // i
  # print(ans)
  for i in range(2, temp+ 1):
    if i in ans:
      print(i, ans.count(i))

for i in range(t):
  solution(int(input()))

